gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-mock/src/main/java/io/gravitee/plugin/endpoint/mock/MockEndpointConnector.java

```diff
@@ -101,7 +101,7 @@ private Flowable<Message> generateMessageFlow(
             .<Message, Long>generate(
                 () -> stateInitValue,
                 (state, emitter) -> {
-                    if (messagesLimitCount == null || (state - stateInitValue) < messagesLimitCount) {
+                    if (messagesLimitCount == null || state < messagesLimitCount) {
```

Comment:
Are you sure about this one? The stateInitValue contains a long and state - stateInitValue allows to actually return the right number of element for the current request. What did it bring you doing this change?

Reply:
Currently it doesn't

Let's take our example with Http-get <-> mock
I make a request with cursor = 9 and the message Limit is 12

`state` starts at 10 and same for `stateInitValue`, you will always be less than `messagesLimitCount` so you will send 12 elements

If I choose a cursor to 9, and the limit is 12, I except to have two messages: 10 and 11